google/cloud/storage/internal/object_write_streambuf.cc

```diff
@@ -107,20 +103,20 @@ std::streamsize ObjectWriteStreambuf::xsputn(char const* s,
   if (!IsOpen()) return traits_type::eof();
 
   auto const actual_size = put_area_size();
-  if (count + actual_size >= max_buffer_size_) {
-    if (actual_size == 0) {
-      FlushRoundChunk({ConstBuffer(s, static_cast<std::size_t>(count))});
-    } else {
-      FlushRoundChunk({
-          ConstBuffer(pbase(), actual_size),
-          ConstBuffer(s, static_cast<std::size_t>(count)),
-      });
-    }
-    if (!last_status_.ok()) return traits_type::eof();
+  if (count + actual_size < max_buffer_size_) {
```

Comment:
Q. Is there any reason to rewrite the test to avoid any possibility of overflow?

For example, if we know `actual_size <= max_buffer_size_`, then we might instead say...
```
  if (count < max_buffer_size_ - actual_size) {
```
Aside: Should the test be `<=`?

Reply:
> Q. Is there any reason to rewrite the test to avoid any possibility of overflow?
> 
> For example, if we know `actual_size <= max_buffer_size_`, then we might instead say...
> 
> ```
>   if (count < max_buffer_size_ - actual_size) {
> ```

Thanks, done.

> 
> Aside: Should the test be `<=`?

I think it should be `<`: we want to flush the data when `count == max_buffer_size - actual_size` because at that point the buffer is large enough to justify the cost of said flush.

Reply:
I see.  It's not "is there space", but rather "is there space plus some".  Thanks.